### PR TITLE
Selecting Between Python 2 and 3

### PR DESCRIPTION
**Pull request by brandjon**
Reviewer TBD. This design impacts native behavior so it'll also be indexed on bazelbuild/proposals and subject to Bazel design review.

### CODE REVIEW

File: proposals/2018-10-25-selecting-between-python-2-and-3.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+reviewers:
+  - TBD
+discussion thread: [bazel #6583](https://github.com/bazelbuild/bazel/issues/6583)
+---
+
+# Selecting Between Python 2 and 3
+
+## Abstract
+
+The "Python mode" configuration value controls whether Python 2 or Python 3 is used to run Python targets built by Bazel. This doc reviews the existing mechanisms for setting the Python mode (the "tri-state model") and describes a simplified mechanism that should replace it (the "boolean model").
```
Comment (aiuto) [inline proposals/2018-10-25-selecting-between-python-2-and-3.md]: nit: doc -> documentation (in the interests of I18N)

Comment (aiuto) [reply proposals/2018-10-25-selecting-between-python-2-and-3.md]: -> document

File: proposals/2018-10-25-selecting-between-python-2-and-3.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+Since there is no longer a third value corresponding to "uncommitted", a target can no longer tell whether it was set to `PY2` mode explicitly (by a flag or a `py_binary`), or if it was set by default because no mode was specified.
+
+### Data dependencies
+
+Since `py_binary` will now change the mode as needed, there is no need to explicitly reset the mode to a particular value (`null`) when crossing `data` attributes. Python 3 targets can freely depend on Python 2 targets and vice versa, so long as the dependency is not via the `deps` attribute in a way that violates `srcs_version` validation (see below).
+
+### Output roots
+
+Since there are only two modes, there need only be two output roots. This avoids action conflicts without any fancy tricks.
```
Comment (aiuto) [inline proposals/2018-10-25-selecting-between-python-2-and-3.md]: Suggestion: This avoids action conflicts in a clear and simple manner.

File: proposals/2018-10-25-selecting-between-python-2-and-3.md (hunk 1 above)
Comment (brandjon) [reply proposals/2018-10-25-selecting-between-python-2-and-3.md]: Done.

**Pull request merged**